### PR TITLE
feat - support additional dynamodb attributes

### PR DESCRIPTION
**Pull request by ghost**
We want to allow using additional dynamoDb attributes. So now the WinstonDynamoDB gets an `additionalAttributesSchema` property as part of the constructor. When a log is sent to the winston-aws-dynamodb, we'll check if any relevant fields from the `additionalAttributesSchema` exist on the log. If they exist we'll write them as part of the dynamo DB object, with the type mentioned in the `additionalAttributesSchema`.

Example of use of additionalAttributesSchema:
```
 return new WinstonDynamoDB({
    tableName: `deployment-step-service-${ENV0_STAGE}-logs-${STAGE}`,
    logStreamName,
    jsonMessage: true,
    uploadRate: 500,
    awsOptions,
    additionalAttributesSchema: {
      deploymentLogId: 'S',
      offsetStart: 'N',
      offsetEnd: 'N'
    }
  });
```

### CODE REVIEW

**Comment by tomer-landesman** [comment]
[add offsets and deploymentLogId to dynamo (winston logger)](https://trello.com/c/ZYh8f84u/8542-add-offsets-and-deploymentlogid-to-dynamo-winston-logger)

**Comment by ghost** [comment]
[add offsets and deploymentLogId to dynamo (winston logger)](https://trello.com/c/ZYh8f84u/8542-add-offsets-and-deploymentlogid-to-dynamo-winston-logger)

**Review by tomer-landesman — APPROVED** [review]
Looks good to me.

File: lib/dynamodb-integration.js [hunk 1]
```diff
@@ -33,35 +35,49 @@ const popEventsToSend = (logEvents, cb) => {
...
-                        }
-                    }
-                })
-            )
-        }
+      RequestItems: {
+        [tableName]: events.map((event) => {
+          const rawMessage = event.rawMessage;
+          const extraValues = {};
+          if (additionalAttributesSchema) {
```
Comment (RLRabinowitz) [inline lib/dynamodb-integration.js]: I think this `if` is redundant. If `additionalAttributesSchema` is not truthy, then `additionalAttributes` is an empty array, and the `forEach` does nothing

**Review by RLRabinowitz — APPROVED** [review]
Looks good! Left one tiny comment

File: lib/dynamodb-integration.js (hunk 1 above)
Comment (ghost) [reply lib/dynamodb-integration.js]: true https://github.com/env0/winston-aws-dynamodb/pull/18/commits/33fba3b5d6587b795a54411f10d16b3b4a47c493

**Pull request merged**